.teamcity/Gradle_Check/projects/StageProject.kt

```diff
@@ -42,20 +45,13 @@ class StageProject(model: CIBuildModel, stage: Stage, containsDeferredTests: Boo
             val functionalTests = FunctionalTestProject(model, testCoverage, stage)
             subProject(functionalTests)
             if (stage.functionalTestsDependOnSpecificBuilds) {
-                functionalTests.buildTypes.forEach { functionalTestBuildType ->
-                    functionalTestBuildType.dependencies {
-                        specificBuildTypes.forEach { specificBuildType ->
-                            dependency(specificBuildType) {
-                                snapshot {
-                                    onDependencyFailure = FailureAction.CANCEL
-                                    onDependencyCancel = FailureAction.CANCEL
-                                }
-                            }
-                        }
-                    }
-
+                specificBuildTypes.forEach { specificBuildType ->
+                    functionalTests.addDependencyForAllBuildTypes(specificBuildType)
```

Comment:
❌ If `dependsOnSanityCheck = then` then the dependency to the SanityCheck build will be created twice. That might cause validation errors. Does the `teamcity-configs:generate` Maven task work correctly?

Reply:
I added a safeguard. Currently, the dependency wouldn't be added twice since `functionalTestsDependOnSpecificBuilds` is never true for our build.